pkg/aws/services/tagging_test.go

```diff
@@ -0,0 +1,421 @@
+				"Key2": aws.String("Value2"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			testName: "exact match with multiple tags",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			testName: "match one tag from source's multiple tags",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+				"Key3": aws.String("Value3"),
+			},
+			testName: "match subnet of tags from source's multiple tags",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key2": aws.String("Value2"),
+			},
+			testName: "no match for tag with different key and value",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value2"),
+			},
+			testName: "no match for tag with same key, different value",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("value1"),
+			},
+			testName: "no match for tag with same key, different value case",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"key1": aws.String("Value1"),
+			},
+			testName: "no match for tag with different key case",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value"),
+			},
+			sourceTags: Tags{},
+			testName:   "no match for tag when source has no tags",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String(""),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			testName: "no match for tag with empty check value",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String(""),
+			},
+			sourceTags: Tags{},
+			testName:   "no match for tag with empty check value when source has no tags",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String(""),
+			},
+			testName: "no match for tag with empty source value",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			assert.Equal(t, containsTags(tt.sourceTags, tt.findTags), tt.contains)
+		})
+	}
+}
+
+func Test_latticeTagging_FindResourcesByTags_nonTargetGroupTypeError(t *testing.T) {
+	c := gomock.NewController(t)
+	ctx := context.TODO()
+	mockLattice := NewMockLattice(c)
+	lt := &latticeTagging{Lattice: mockLattice}
+
+	nonTargetGroupResourceType := ResourceTypeService
+
+	_, err := lt.FindResourcesByTags(ctx, nonTargetGroupResourceType, Tags{})
+	assert.NotNil(t, err)
```

Comment:
This is fine, but I think `assert.Error` is a bit more idiomatic.

Reply:
Done